extensions/netty-loom-adaptor/deployment/src/main/java/io/quarkus/netty/loom/adaptor/NettyLoomAdaptorProcessor.java

```diff
@@ -0,0 +1,670 @@
+package io.quarkus.netty.loom.adaptor;
+
+import static org.objectweb.asm.Opcodes.*;
+
+import java.io.IOException;
+import java.util.function.BiFunction;
+
+import org.jboss.jandex.DotName;
+import org.jboss.logging.Logger;
+import org.objectweb.asm.*;
+
+import io.quarkus.builder.item.*;
+import io.quarkus.builder.item.EmptyBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.Consume;
+import io.quarkus.deployment.annotations.Produce;
+import io.quarkus.deployment.builditem.BytecodeTransformerBuildItem;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.gizmo.Gizmo;
+import io.quarkus.netty.deployment.MinNettyAllocatorMaxOrderBuildItem;
+import io.smallrye.common.annotation.RunOnVirtualThread;
+
+public class NettyLoomAdaptorProcessor {
```

Comment:
I think we'll need some comments on what this does, otherwise it will be hard to maintain this ASM :)

Reply:
It would definitely make sense to display the actual result of the transformation in addition to why we need it.

Reply:
hey @geoand, I added some comments explaining what we do
If you find it satisfying I'll just squash everything and rebase to make it merge-ready

Reply:
Great, thanks